src/Core/System/Snippet/SnippetService.php

```diff
@@ -122,11 +126,31 @@ public function getStorefrontSnippets(MessageCatalogueInterface $catalog, string
             $locale === $fallbackLocale ? $fallbackSnippets : $this->getSnippetsByLocale($snippetCollection, $locale)
         );
 
-        // at least overwrite the snippets with the database customer overwrites
-        return array_replace_recursive(
+        $event = $this->eventDispatcher->dispatch(new StorefrontSnippetsAfterCurrentLocaleEvent(
             $snippets,
+            $locale,
+            $catalog,
+            $snippetSetId,
+            $fallbackLocale,
+            $salesChannelId
+        ));
+
+        // at least overwrite the snippets with the database customer overwrites
+        $snippets = array_replace_recursive(
+            $event->snippets,
             $this->fetchSnippetsFromDatabase($snippetSetId, $unusedThemes)
         );
+
+        $event = $this->eventDispatcher->dispatch(new StorefrontSnippetsAfterDatabaseOverwritesEvent(
```

Comment:
```suggestion
        $event = $this->eventDispatcher->dispatch(new StorefrontSnippetsAfterDatabaseOverrideEvent(
```
It's just a small one, but this should be "Override" (non destructive) instead of "Overwrite" (destructive)

Reply:
I switched from Event to Extension.